hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/config/RateLimitConfiguration.java

```diff
@@ -0,0 +1,47 @@
+package com.hedera.mirror.web3.config;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2023 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import io.github.bucket4j.Bandwidth;
+import io.github.bucket4j.Bucket;
+import io.github.bucket4j.Refill;
+import java.time.Duration;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import com.hedera.mirror.web3.evm.properties.MirrorNodeEvmProperties;
+
+@Configuration
+@RequiredArgsConstructor
+class RateLimitConfiguration {
+    private final MirrorNodeEvmProperties mirrorNodeEvmProperties;
+
+    @Bean
+    Bucket buildBucket() {
+        final var rateLimitPerSecond = mirrorNodeEvmProperties.getRateLimitPerSecond();
+        final var limit = Bandwidth.classic(rateLimitPerSecond, Refill.greedy(10, Duration.ofSeconds(1)));
```

Comment:
Does this actually limit the rate to 10 rps and allow a burst of rateLimitPerSecond in case rateLimitPerSecond > 10?

My understanding is, if we want to have a rate limit per the configure, it should be:

```suggestion
        final var limit = Bandwidth.classic(rateLimitPerSecond, Refill.greedy(rateLimitPerSecond, Duration.ofSeconds(1)));
```

Reply:
Hey, At first the refill rate of tokens was with the `rateLimitPerSecond`(100), but when measuring the performance we noticed that the RPS is often above 100 (there were values of 258+ RPS). With the change from 100 to 10 the RPS measure never gets above 99. The measure is done thru the existing k6 tests.

Reply:
no idea why perf tests show the numbers you mentioned. however, this is [the official doc](https://bucket4j.com/7.6.0/toc.html#limiting-the-rate-of-access-to-rest-api). in the example

```java
    private Bucket createNewBucket() {
         long overdraft = 50;
         Refill refill = Refill.greedy(10, Duration.ofSeconds(1));
         Bandwidth limit = Bandwidth.classic(overdraft, refill);
         return Bucket.builder().addLimit(limit).build();
    }
```

This is for a throttle of 10 calls/sec with a short period burst controlled by the `overdraft`

Reply:
Perhaps just switch to `Bandwidth.simple()` if we don't understand the harder to use classic mode?

Reply:
agree, especially if we don't think burst / overdraft is required, `simple()` is just

```
    public static Bandwidth simple(long capacity, Duration period) {
        Refill refill = Refill.greedy(capacity, period);
        return classic(capacity, refill);
    }
```